app/controllers/api/database.php

```diff
@@ -1640,16 +1640,19 @@ function createAttribute($collectionId, $attribute, $response, $dbForInternal, $
         $data['$read'] = (is_null($read) && !$user->isEmpty()) ? ['user:'.$user->getId()] : $read ?? []; //  By default set read permissions for user
         $data['$write'] = (is_null($write) && !$user->isEmpty()) ? ['user:'.$user->getId()] : $write ?? []; //  By default set write permissions for user
 
-        // Users can only add their roles to documents, API keys can add any
+        // Users can only add their roles to documents, API keys and Admin users can add any
         $roles = \array_fill_keys(Authorization::getRoles(), true); // Auth::isAppUser expects roles to be keys, not values of assoc array
-        foreach ($data['$read'] as $read) {
-            if (!Auth::isAppUser($roles) && !Authorization::isRole($read)) {
-                throw new Exception('Read permissions must be one of: ('.\implode(', ', $roles).')', 400);
+
+        if (!Auth::isAppUser($roles) && !Auth::isPrivilegedUser($roles)) {
+            foreach ($data['$read'] as $read) {
+                if (!Authorization::isRole($read)) {
+                    throw new Exception('Read permissions must be one of: ('.\implode(', ', array_keys($roles)).')', 400);
+                }
             }
-        }
-        foreach ($data['$write'] as $write) {
-            if (!Auth::isAppUser($roles) && !Authorization::isRole($write)) {
-                throw new Exception('Write permissions must be one of: ('.\implode(', ', $roles).')', 400);
+            foreach ($data['$write'] as $write) {
+                if (!Authorization::isRole($write)) {
+                    throw new Exception('Write permissions must be one of: ('.\implode(', ', array_keys($roles)).')', 400);
```

Comment:
We should probably fix this instead of hacking around the data structure mismatch 😆 
The output of `Authorization::getRoles()` should match the input of `Auth::isAppUser()` - an indexed array is probably the easiest instead of associative.

Reply:
Done 🥳